.github/workflows/nodejs.yml

```diff
@@ -9,214 +9,276 @@ concurrency:
-      - name: build
-        if: steps.build-cache.outputs.cache-hit != 'true'
-        run: yarn build:packages --diff --deps
-      - name: tests
-        run: yarn test --diff --deps --no-cache --ci --runInBand --reporters=default --reporters=jest-junit
-        env:
-          JEST_JUNIT_OUTPUT: 'reports/junit/js-test-results.xml'
+          node-version: '14'
+          cache: yarn
+          cache-dependency-path: 'yarn.lock'
 
-  tests-ssr:
-    name: Run tests with artificial server rendering
+      - run: yarn install --frozen-lockfile --prefer-offline
+
+      - name: prettier
+        run: yarn check:prettier
+
+      - name: eslint
+        run: yarn check:eslint --format junit -o reports/junit/js-lint-results.xml
+
+  types:
+    name: Check types
     runs-on: ubuntu-latest
+    needs: install
     steps:
-      - name: Checkout Repo
-        uses: actions/checkout@v2
-      - name: Use Node.js 14.x
-        uses: actions/setup-node@v1
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 14
+        uses: actions/setup-node@v3
         with:
-          node-version: '14.x'
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-      - uses: actions/cache@v2
-        id: yarn-cache
+          node-version: '14'
+          cache: yarn
+          cache-dependency-path: 'yarn.lock'
+
+      - run: yarn install --frozen-lockfile --prefer-offline
+
+      - name: types
+        run: yarn check:ts
+
+  deps:
+    name: Check dependencies
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 14
+        uses: actions/setup-node@v3
         with:
-          path: |
-            ${{ steps.yarn-cache-dir-path.outputs.dir }}
-            **/node_modules
-          key: ${{ runner.os }}-yarn-cache-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-cache-
-      - run: yarn install --frozen-lockfile
-      - uses: actions/cache@v2
+          node-version: '14'
+          cache: yarn
+          cache-dependency-path: 'yarn.lock'
+
+      - run: yarn install --frozen-lockfile --prefer-offline
+
+      - name: deps
+        run: yarn check:deps
+
+      - name: package.json versions
+        run: yarn check:package-json
+
+  build:
+    name: Build packages
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v3
+        name: Check for build cache
         id: build-cache
         with:
           path: |
             **/dist
             !**/node_modules
           key: ${{ runner.os }}-build-cache-${{ hashFiles('**/rollup.config.js', '**/babel.config.js', '**/src/', '**/scripts/', '!**/node_modules', '!**/dist') }}
+
+      # Only setup & build if there was no build cache hit
+      - name: Use Node.js 14
```

Comment:
will there ever be a cache hit at this point in the workflow? i thought github actions didn't persist caches between workflows, but i could definitely be wrong about that

Reply:
nvm i think i am wrong

Reply:
but i do think we need to give this cache a key?

Reply:
i thoguht `key` was required: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action